tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/validation/PlacementValidator.java

```diff
@@ -204,4 +207,26 @@ private void checkSpecialtyType(final List<FieldError> fieldErrors,
           String.format("Only one Specialty of type %s allowed", PostSpecialtyType.SUB_SPECIALTY)));
     }
   }
+
+  private List<FieldError> checkNpnUpdateIsAllowed(final PlacementDetailsDTO placementDetailsDto) {
+    final List<FieldError> fieldErrors = new ArrayList<>();
+
+    Optional<Placement> dbPlacement = placementRepository.findPlacementById(
+        placementDetailsDto.getId());
+
+    if (!dbPlacement.isPresent()) {
+      return fieldErrors;
+    }
+
+    if (!dbPlacement.get().getPlacementEsrEvents().isEmpty()) {
+      String oldNpn = dbPlacement.get().getPost().getNationalPostNumber();
+      String newNpn = placementDetailsDto.getNationalPostNumber();
+      if (!StringUtils.equals(newNpn, oldNpn)) {
```

Comment:
I combined the two conditions and pushed it. However this means that oldNpn and newNpn will always be evaluated, rather than be skipped altogether when the placement has no esrEvents.

Reply:
I think this should not be worried because you've ready got the old placement from the DB and there's no extra cost to get the NPNs. 
Another one I noticed is `dbPlacement.get().getPost()` could return `null` in theory, but it should be fine here because all of our placements have a post